cli/azd/pkg/tools/azcli/container_registry.go

```diff
@@ -23,23 +25,26 @@ type ContainerRegistryService interface {
 }
 
 type containerRegistryService struct {
-	userAgent  string
-	httpClient httputil.HttpClient
-	docker     docker.Docker
-	credential azcore.TokenCredential
+	userAgent    string
+	httpClient   httputil.HttpClient
+	docker       docker.Docker
+	subResolver  account.SubscriptionTenantResolver
+	credProvider auth.MultiTenantCredentialProvider
 }
 
 // Creates a new instance of the ContainerRegistryService
 func NewContainerRegistryService(
-	credential azcore.TokenCredential,
+	subResolver account.SubscriptionTenantResolver,
+	credProvider auth.MultiTenantCredentialProvider,
```

Comment:
> This feels wrong that we now have to pass to services around to anything that needs credentials to realize our final credentials for the service. In my opinion we should either take the credentials object as it was or have a single service that we can rely out to generate the credentials that we need. Having each service generate its own azure credential feels like an anti-pattern to me.

I think the fundamental issue here is that we need to vary the credential that we use depending on what subscription we are targeting for a specific operation, since the subscription belongs to a tenant, and we need to ensure we use that tenant id in our credential.

The other option would be to pass the credential and subscription when we construct the object and keep them fixed for the life of the object.  This also doesn't feel like it will play well with our IoC system.